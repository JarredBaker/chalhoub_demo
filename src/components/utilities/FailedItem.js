import React from 'react';
import {Text, StyleSheet} from 'react-native';
import propTypes from 'prop-types';

/**
 *
 * Component used to display an error message and view
 * @function FailedItem(@param {String} text the message on the screen)
 *
 *
 */
export default FailedItem = ({text}) => {
  return <Text style={styles.errorText}>{text}</Text>;
};

const styles = StyleSheet.create({
  errorText: {
    fontSize: 18,
    textAlign: 'center',
    color: 'red',
  },
});

FailedItem.propTypes = {
  text: propTypes.string.isRequired,
};
