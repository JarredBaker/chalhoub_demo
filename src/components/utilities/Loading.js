import React from 'react';
import {Text, StyleSheet} from 'react-native';

/**
 *
 * Component used to display loading.
 * @function Loading
 *
 *
 */
export default Loading = () => {
  return <Text style={styles.loadingText}>Loading...</Text>;
};

const styles = StyleSheet.create({
  loadingText: {
    fontSize: 28,
    textAlign: 'center',
  },
});
