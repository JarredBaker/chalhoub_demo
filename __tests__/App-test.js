/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '../src/components/utilities/Loading';
import FailedItem from '../src/components/utilities/FailedItem';
jest.mock('axios');

it(`renders loading item correctly`, () => {
  const tree = renderer.create(<Loading />);
  expect(tree).toMatchSnapshot();
});

it(`renders failed item correctly`, () => {
  const treeFailed = renderer.create(
    <FailedItem text={'Please try again later'} />,
  );
  expect(treeFailed).toMatchSnapshot();
});
