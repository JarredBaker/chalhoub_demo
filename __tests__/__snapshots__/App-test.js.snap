// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Object {
      "fontSize": 28,
      "textAlign": "center",
    }
  }
>
  Loading...
</Text>
`;

exports[`renders correctly 2`] = `
<Text
  style={
    Object {
      "color": "red",
      "fontSize": 18,
      "textAlign": "center",
    }
  }
/>
`;

exports[`renders failed item correctly 1`] = `
<Text
  style={
    Object {
      "color": "red",
      "fontSize": 18,
      "textAlign": "center",
    }
  }
>
  Please try again later
</Text>
`;

exports[`renders loading item correctly 1`] = `
<Text
  style={
    Object {
      "fontSize": 28,
      "textAlign": "center",
    }
  }
>
  Loading...
</Text>
`;
